### ISSUE TITLE
inconsistent-return-statements and functions which never return

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#1771]
On pylint 1.8.1, astroid 1.6.0 with this file:

```python
import sys

def foo(n):
    if n == 1:
        sys.exit(1)
    else:
    	return n * 2
```

I get `inconsistent-return-statements`, despite there being no way `sys.exit` will return. Arguably here I could just get rid of the `else:`, but there are other similar cases, and doing an explicit `raise` above instead makes the warning go away.

Maybe there should be a setting for fully-qualified names of functions which will never return, with `sys.exit` being the default (maybe more)?

**Comment by hippo91** [comment]
You are right again @The-Compiler!
This new checker seems to need enhancement...
